ToDoApp/ToDoApp/ToDoService.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  ToDoDetail.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-11-29.
+//
+
+import Foundation
+
+enum TaskError: Error {
+    case unavailableTask
+}
+
+struct Task {
+    var id: String = ""
+    var name: String = ""
+    var deadline: Date = Date.now
+    var status: Bool = false
+    init(name: String, deadline: Date?) {
+        self.name = name
+        self.deadline = deadline!
+    }
+}
+
+protocol ToDoService {
+    func createTask(name: String, deadline: Date?) -> Task
+    func updateTask(todo: Task) throws
+    func taskForToday() -> [Task]
```

Comment:
maybe you would like to introduce async interfaces here, since when you will connect a cloud service you won't refactor client code which already use ToDoService

you can choose either old fashion closure interfaces or new async interfaces
`func createTask(name: String, deadline: Date?, completion: @escaping (Result<Task, TaskError>) -> Void)`
or
`func createTask(name: String, deadline: Date?) async -> Result<Task, TaskError>`

Reply:
This has been fixed.